Sources/MiniPlayer/AudioMiniPlayer.swift

```diff
@@ -0,0 +1,168 @@
+
+extension AudioMiniPlayer {
+    func prepare(forMediaPlayback controller: VLCPlaybackController) {
+        updatePlayPauseButton()
+        controller.delegate = self
+        controller.recoverDisplayedMetadata()
+        // For now, AudioMiniPlayer will be used for all media
+        controller.videoOutputView = artworkImageView
+    }
+
+    func mediaPlayerStateChanged(_ currentState: VLCMediaPlayerState,
+                                 isPlaying: Bool,
+                                 currentMediaHasTrackToChooseFrom: Bool,
+                                 currentMediaHasChapters: Bool,
+                                 for controller: VLCPlaybackController) {
+        updatePlayPauseButton()
+    }
+
+    func displayMetadata(for controller: VLCPlaybackController, metadata: VLCMetaData) {
+        setMediaInfo(metadata)
+    }
+
+    func playbackPositionUpdated(_ controller: VLCPlaybackController) {
+        progressBarView.progress = controller.playbackPosition
+    }
+}
+
+// MARK: - UI Receivers
+
+private extension AudioMiniPlayer {
+    @IBAction private func handlePrevious(_ sender: UIButton) {
+        VLCPlaybackController.sharedInstance().previous()
+    }
+
+    @IBAction private func handlePlayPause(_ sender: UIButton) {
+        VLCPlaybackController.sharedInstance().playPause()
+        updatePlayPauseButton()
+    }
+
+    @IBAction private func handleNext(_ sender: UIButton) {
+        VLCPlaybackController.sharedInstance().next()
+    }
+
+    @IBAction private func handleFullScreen(_ sender: Any) {
+        UIApplication.shared.sendAction(#selector(VLCPlayerDisplayController.showFullscreenPlayback),
+                                        to: nil,
+                                        from: self,
+                                        for: nil)
+    }
+
+    @IBAction private func handleSwipe(_ sender: UISwipeGestureRecognizer) {
+        switch sender.direction {
+        case .right:
+            VLCPlaybackController.sharedInstance().previous()
+        case .left:
+            VLCPlaybackController.sharedInstance().next()
+        default:
+            break
+        }
+    }
+
+    @IBAction private func handleLongPressPlayPause(_ sender: UILongPressGestureRecognizer) {
+        switch sender.state {
+        // case .began:
+        // In the case of .began we could a an icon like the old miniplayer
+        case .ended:
+            VLCPlaybackController.sharedInstance().stopPlayback()
+        case .cancelled, .failed:
+            VLCPlaybackController.sharedInstance().playPause()
+            updatePlayPauseButton()
+        default:
+            break
+        }
+    }
+}
+
+// MARK: - Setters
+
+private extension AudioMiniPlayer {
+    private func setMediaInfo(_ metadata: VLCMetaData) {
+        artworkImageView.image = metadata.artworkImage ?? UIImage(named: "no-artwork")
+        titleLabel.text = metadata.title
+        artistLabel.text = metadata.artist
```

Comment:
should this now also handle video ? we might need to show different values for that if it's a video

Reply:
Not with the adaptation design, there is nothing more than the title.

Though, I will add a check to avoid a useless assignment when it is a video.